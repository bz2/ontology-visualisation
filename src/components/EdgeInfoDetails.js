import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import actions from '../store/actions'
import EdgeInfoDetailsRow from './EdgeInfoDetailsRow'

const EdgeInfoDetails = ({
  edgeId,
  objectPropertiesFromApi,
  edgesConnections,
}) => {
  const { t } = useTranslation()

  const { OwlObjectProperties } = objectPropertiesFromApi

  const selectedEdge = OwlObjectProperties[edgeId]

  const tableRowNames = Object.keys(selectedEdge).filter((key) => typeof selectedEdge[key] !== 'object'
    && !key.includes('label')).sort()

  let connections = []

  if (edgesConnections[edgeId]) {
    connections = edgesConnections[edgeId]
  }

  return (
    <div className="edge-info-details">
      <div className="edge-info-details-title">
        {t('properties')}
      </div>

      <div className="edge-info-details-table">
        <table>
          <thead>
            <tr>
              <th />
              <th />
            </tr>
          </thead>
          <tbody>
            {
              tableRowNames.map((tableRowName) => (
                <tr
                  key={`details-row-${tableRowName}`}
                >
                  <td>
                    {tableRowName}
                  </td>
                  <td>
                    {selectedEdge[tableRowName] || t('null')}
                  </td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>

      <div className="edge-info-details-title">
        {t('nodesProperties')}
      </div>

      <div className="edge-info-details-table">
        <table>
          <thead>
            <tr>
              <th />
              <th>{t('from')}</th>
              <th>{t('to')}</th>
            </tr>
          </thead>
          <tbody>
            {
              connections.map((connection) => (
                <EdgeInfoDetailsRow
                  key={`edge-relationship-row-${connection.from}-${connection.to}`}
                  connection={connection}
                />
              ))
            }
          </tbody>
        </table>
      </div>
    </div>
  )
}

EdgeInfoDetails.propTypes = {
  edgeId: PropTypes.string.isRequired,
  objectPropertiesFromApi: PropTypes.shape().isRequired,
  edgesConnections: PropTypes.shape().isRequired,
}

const mapToProps = ({
  objectPropertiesFromApi,
  nodesConnections,
  edgesConnections
}) => ({
  objectPropertiesFromApi,
  nodesConnections,
  edgesConnections
})

export default connect(
  mapToProps,
  actions
)(EdgeInfoDetails)
