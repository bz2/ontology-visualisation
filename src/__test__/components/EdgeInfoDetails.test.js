import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EdgeInfoDetails from '../../components/EdgeInfoDetails'

const setup = () => {
  const props = {
    edgeId: 'abc',
    objectPropertiesFromApi: {
      OwlObjectProperties: {
        abc: {
          id: 'abc',
          rdfsLabel: 'label',
        }
      }
    },
    edgesConnections: [{
      from: 'abc',
      fromLabel: 'label',
      to: 'cde',
      toLabel: 'label2',
    }],
  }

  const component = shallow(<EdgeInfoDetails {...props} />)

  return {
    component,
    props
  }
}

describe('EdgeInfoDetails', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
