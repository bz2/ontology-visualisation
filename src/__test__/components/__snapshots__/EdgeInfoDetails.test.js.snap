// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgeInfoDetails should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
      "updateGraphData": [Function],
    }
  }
  edgeId="abc"
  edgesConnections={
    Array [
      Object {
        "from": "abc",
        "fromLabel": "label",
        "to": "cde",
        "toLabel": "label2",
      },
    ]
  }
  mapToProps={[Function]}
  objectPropertiesFromApi={
    Object {
      "OwlObjectProperties": Object {
        "abc": Object {
          "id": "abc",
          "rdfsLabel": "label",
        },
      },
    }
  }
>
  <Component />
</Connect>
`;
